hooks/pre_validations/30-mongo_storage_engine.rb

```diff
@@ -0,0 +1,47 @@
+MONGO_DIR = '/var/lib/mongodb/'
+
+def fail_and_exit(message)
+  Kafo::Helpers.log_and_say :error, message
+  kafo.class.exit 1
+end
+
+def disk_space
+  # Check diskspace in /var/tmp
+  logger.info 'Checking available diskspace in /var/tmp for upgrade'
+  total_space = `df -k --output=avail /var/tmp`.split("\n").last.to_i
+  mongo_size = File.directory?(MONGO_DIR) ? `du -s  #{@MONGO_DIR}`.split[0].to_i : 0
+  if total_space < mongo_size
+    fail_and_exit "There is not enough free space #{total_space}, the size of MongoDB database is #{mongo_size}, please add additional space to /var/tmp and try again, exiting."
```

Comment:
I think this will fail if it is symlinked to another dir in order to give it more disk space. I might want to do this if don't want to increase the partition `/var/tmp` is on to upgrade.

I'm not if we care about this scenario?

Reply:
Then want would be your suggested method to upgrade if I have no free space available to increase partitions or create a new partition but I do have space already allocated to another partition that I could use?

Reply:
We can add an option to use the --disable-system-checks, I don't want to keep changing this piece around too much since we want to keep track of most use cases.